api/v1alpha1/clusterconfig_types.go

```diff
@@ -344,6 +348,71 @@ func (ImageRegistries) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type Users []User
+
+func (Users) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Users to add to the machine",
+			Type:        "array",
+			Items:       ptr.To(User{}.VariableSchema().OpenAPIV3Schema),
+		},
+	}
+}
+
+// User defines the input for a generated user in cloud-init.
+type User struct {
+	// Name specifies the user name
+	Name string `json:"name"`
+
+	// HashedPassword specifies a hashed password for the user. See your
+	// distribution's documentation for instructions to create a hashed
+	// password.
+	// An empty string is not marshalled, because it is not a valid value.
+	// +optional
+	HashedPassword string `json:"hashedPassword,omitempty"`
+
+	// SSHAuthorizedKeys specifies a list of public SSH keys to write to the
+	// machine. Use the corresponding private SSH keys to authenticate. See SSH
+	// documentation for instructions to create a key pair.
+	// +optional
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// Sudo specifies a sudo role for the user
+	// +optional
+	Sudo *string `json:"sudo,omitempty"`
+}
+
+func (User) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Type: "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
```

Comment:
Let's specify at least name as required:

```suggestion
			Required: []string{"name"},
			Properties: map[string]clusterv1.JSONSchemaProps{
```